### DISCUSSION TITLE
Sourced

### DISCUSSION THREAD

**Discussion by dieBakterie** [prasanthrangan/hyprdots#1457] [Ideas]
Is it possible to move to an more sourced attempt or structure, like this one for example?
```
 .config/hypr
├──  animations.conf
├──  environments.conf
├──  hypridle.conf
├──  hyprland.conf
├──  hyprland_sourced.conf
├──  hyprlock.conf
├──  input.conf
├──  keybindings.conf
├──  keybindings_sourced.conf
├──  launch.conf
├──  layouts.conf
├──  misc.conf
├──  monitors.conf
├──  monitors_sourced.conf
├──  nvidia.conf
├──  scratchpads.conf
├──  themes
│  ├──  colors.conf
│  ├──  common.conf
│  └──  theme.conf
├──  userprefs.conf
├──  vm.conf
├──  windowrules.conf
└──  workspaces.conf
```
The configs are self explaining i guess. In `layout.conf` are the layout section from the `hyprland.conf`, in input the input section and so on. In `vm.conf` stands the VM example from within `userprefs.t2`, the source for `scratchpass.conf` is commented Out in `hyprland_sourced.conf`, i've two Sessions, Hyde and sourced and the `workspaces.conf` is empty right now. Same goes for `monitors.conf` and `monitors_sourced.conf` the **_sourced** is the actual one, the other is the _"same"_. Doesn't wanted to overwrite those upstream that's why i've those Files twice. In `launch.conf` are the Launch section from the `hyprland.conf`, in `environments.conf` are the environment variables and the order in what you source those Files are kinda important for it to work properly and you should avoid empty lines, like this
```
#// test
#// keybindings
```
just comment those out. It can lead to errors(an empty Line gets marked as invalid), don't know why, isn't my fault, i guess it's Hyprland's but why idk. That's why you should avoid empty lines in your configs. For reference i've an fork and within the hyDE Branch is the sourced Version, just ignore the missing `userprefs.conf`, don't really need it added my userprefs in the other Files. The hyprlock Implementation is WIP and don't really Work right now.

**Comment by kRHYME7** [comment]
and hyprlock should not be sourced in case you don't know

**Reply by dieBakterie** [reply]
Thanks for the Info and if you want i can make a PR later for this, sourced Hyprland only obviously.

**Comment by dieBakterie** [comment]
![240509_16h15m03s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/50206827/5099ed39-e548-4c90-95ca-34520918fe2f)
That's what i meant with unfaulty errors don't know why this happens.

**Reply by dieBakterie** [reply]
If i comment out the line or remove it it's okay, why an empty line throws this error no idea. I'll open an issue on Hyprland for this.
